lib/ruby_lsp/document.rb

```diff
@@ -83,6 +84,11 @@ def parsed?
       !@tree.nil?
     end
 
+    sig { returns(Scanner) }
+    def scanner
```

Comment:
Nitpick: Since it initialises a new `Scanner` every time it's called, may be add a verb prefix, like `create_scanner`?
Otherwise, based on its usages I'd assume it's an attribute of the document.